synapse/handlers/admin.py

```diff
@@ -426,9 +429,18 @@ async def _redact_all_events(
         user_id = task.params.get("user_id")
         assert user_id is not None
 
-        # puppet the user if they're ours, otherwise use admin to redact
+        use_admin = task.params.get("use_admin")
+        assert use_admin is not None
```

Comment:
```suggestion
        use_admin = task.params.get("use_admin", False)
```

These are stored in the DB and persist across restarts, so we may have some pending tasks that don't have this flag set yet